prime-router/docs/design/design/deduplication/deduplication.md

```diff
@@ -0,0 +1,197 @@
+
+```json
+{
+    "id": "849b3151-25f7-41f3-b19b-fd8ad47bae18",
+    "submissionId": 63,
+    "overallStatus": "Delivered",
+    "timestamp": "2025-01-22T00:42:10.404Z",
+    "sender": "development.dev-elims",
+    "reportItemCount": 2,
+    "errorCount": 0,
+    "warningCount": 1,
+    "httpStatus": 201,
+    "destinations": [
+        {
+            "organization": "FOR DEVELOPMENT PURPOSES ONLY",
+            "organization_id": "development",
+            "service": "DEV_ENRICHMENT_FHIR",
+            "itemCount": 1,
+            "itemCountBeforeQualityFiltering": 0,
+            "sending_at": "2025-01-22T00:43:00.000Z",
+            "filteredReportRows": [],
+            "filteredReportItems": [],
+            "sentReports": [
+                {
+                    "reportId": "dbc9ac68-77a0-4e3e-8d8b-2d665b8fd92c",
+                    "externalName": "none-c7c7e0fe-1951-4031-9d17-2895fc03c974-20250121164303.fhir",
+                    "createdAt": "2025-01-22T00:43:04.252Z",
+                    "itemCount": 1
+                }
+            ],
+            "downloadedReports": []
+        }
+    ],
+    "reportId": "849b3151-25f7-41f3-b19b-fd8ad47bae18",
+    "topic": "full-elr",
+    "errors": [],
+    "warnings": [
+        {
+            "scope": "item",
+            "indices": [
+                1
+            ],
+            "trackingIds": [
+                "849b3151-25f7-41f3-b19b-fd8ad47bae18"
+            ],
+            "message": "Report could not processed. Reason: Duplicate item was detected and removed.",
+            "errorCode": "DUPLICATE_ITEM"
+        }
+    ],
+    "destinationCount": 1,
+}
+```
+
+##### Example: Submission History API, Entire Report is Duplicate
+
+```json
+{
+    "id": "849b3151-25f7-41f3-b19b-fd8ad47bae18",
+    "overallStatus": "Not Delivered",
+    "timestamp": "2025-01-22T00:42:10.404Z",
+    "sender": "development.dev-elims",
+    "reportItemCount": 1,
+    "errorCount": 1,
+    "warningCount": 0,
+    "httpStatus": 201,
+    "destinations": [],
+    "reportId": "849b3151-25f7-41f3-b19b-fd8ad47bae18",
+    "topic": "full-elr",
+    "errors": [
+        {
+            "scope": "item",
+            "indices": [
+                1
+            ],
+            "trackingIds": [
+                "849b3151-25f7-41f3-b19b-fd8ad47bae18"
+            ],
+            "message": "Report could not processed. Reason: Duplicate report was detected and removed.",
+            "errorCode": "DUPLICATE_REPORT"
+        }
+    ],
+    "warnings": [],
+    "destinationCount": 0,
+}
+```
+
+### Deduplication Workflow Placement
+
+<BR><img alt="Dedupe_Design.jpg" height="310" src="Dedupe_Design.jpg" width="450"/>
+
+- The deduplication workflow will happen at the very end of the convert step processing ([FHIRConverter.kt#L541](https://github.com/CDCgov/prime-reportstream/blob/c942a9a6f6be347d82196939e1cf677512af4a06/prime-router/src/main/kotlin/fhirengine/engine/FHIRConverter.kt#L541)).
```

Comment:
According to your diagram it does not happen "at the very end"

Reply:
I'll update the wording here. This is referring to the FHIRConverter method process() which does the everything listed before deduplication. After that method there are still FHIRConverter actions taken but they're more related to informational actions (updating the db, action logging, azure events, queueing, etc).